movieController.go

```diff
@@ -37,13 +45,304 @@ func moviesByName(c *gin.Context) {
+				result := db.Save(&entry)
+				if result.Error != nil {
+					log("ERROR", "Failed to post deletion to postgres")
+					c.JSON(http.StatusBadRequest, Response{Success: false, Context: result.Error.Error()})
+				} else {
+					log("INFO", "Deleted user")
+					c.JSON(http.StatusOK, Response{Success: true, Context: "Deleted User"})
+				}
+			}
+		} else {
+			log("ERROR", "Attempted to delete an inactive account")
+			c.JSON(http.StatusBadRequest, Response{Success: false, Context: "Please enter a valid user ID"})
+		}
+	}
+}
+
+// updateUser godoc
+// @Summary updateUser
+// @Description update a user with a User-Id and new username and password
+// @Tags users
+// @Param User-Id header string true "UserID" 
+// @Param UserData body Users true "User Data"
+// @Accept */*
+// @Produce json
+// @Router /updateUser [PATCH]
+func updateUser(c *gin.Context) {
+	checkTime()
+	header := c.Request.Header["User-Id"]
+	if len(header) == 0 {
+		log("ERROR", "Invalid User-Id given")
+		c.JSON(http.StatusBadRequest, Response{Success: false, Context: "Please enter a valid user ID"})
+	} else {
+		body, err := c.GetRawData()
+		if err != nil {
+			log("ERROR", "Failed to read request body")
+			c.JSON(http.StatusBadRequest, Response{Success: false, Context: "Failed to read request body"})
+		} else {
+			var entry User
+			result := db.Where("id = ?", string(header[0])).First(&entry)
+			if entry.Active {
+				if result.Error != nil {
+					log("ERROR", "Invalid User-Id given")
+					c.JSON(http.StatusBadRequest, Response{Success: false, Context: "Please enter a valid user ID"})
+				} else {
+					var creds Users
+					json.Unmarshal(body, &creds)
+					fmt.Println(creds)
+					hash, err := bcrypt.GenerateFromPassword([]byte(creds.Password), bcrypt.DefaultCost)
+					if err != nil {
+						log("ERROR", "Failed to encrypt")
+						c.JSON(http.StatusBadRequest, Response{Success: false, Context: "Failed to encrypt"})
+					} else {
+						entry.Username = creds.Username
+						entry.Password = string(hash)
+						entry.UpdatedAt = time.Now().String()
+						result := db.Save(&entry)
+						if result.Error != nil {
+							log("ERROR", "Could not post user info to postgres")
+							c.JSON(http.StatusBadRequest, Response{Success: false, Context: result.Error.Error()})
+						} else {
+							log("INFO", "Updated user info")
+							c.JSON(http.StatusOK, Response{Success: true, Context: "Updated info"})
+						}
+					}
+				}
+			} else {
+				log("WARNING", "Attempted to update inactive account")
+				c.JSON(http.StatusBadRequest, Response{Success: false, Context: "Please enter a valid user ID"})
+			}
+
+		}
+	}
+}
+
+// finishGame godoc
+// @Summary finishGame
+// @Description update success rates of user and daily
+// @Tags users
+// @Param User-Id header string false "UserID" 
+// @Param UserData body Users true "User Data"
+// @Accept */*
+// @Produce json
+// @Router /finishGame [POST]
+func finishGame(c *gin.Context) {
+	checkTime()
```

Comment:
do we need to be calling checkTime on every single api call? Would it make sense to only have this on MoviesByname?

Reply:
Good catch